tests/test_qsim_device.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
```

Comment:
The tests in this file have been copied from `test_cirq_device.py`, with only some minor modifications to work with `QSimDevice`.